cassandra/src/main/java/org/apache/calcite/adapter/cassandra/CassandraFilter.java

```diff
@@ -174,14 +174,13 @@ private String translateMatch(RexNode condition) {
       }
     }
 
-    /** Convert the value of a literal to a string.
+    /** Returns the value of the literal.
      *
      * @param literal Literal to translate
-     * @return String representation of the literal
+     * @return The value of the literal in the form of the actual type.
      */
-    private static String literalValue(RexLiteral literal) {
-      Object value = literal.getValue2();
-      return String.valueOf(value);
+    private static Object literalValue(RexLiteral literal) {
+      return literal.getValue2();
     }
```

Comment:
For `Integral` and `string` it is correct, but i'm not sure if it is for other data types literal, such as `TIMESTAMP(3)`, can you make sure that ?

Reply:
> For `Integral` and `string` it is correct, but i'm not sure if it is for other data types literal, such as `TIMESTAMP(3)`, can you make sure that ?

Thanks a lot for reviewing. This PR has been updated.
I have tested for type like Date, Timestamp and Time.
1.When the type of primary key 'f_time' is Time, it will throw exception when the filter condition is "f_time = Time '13:30:54.234'", because the type of 'f_time' is Bigint.
It seems another issue.
2.As for Timestamp(3), CassandraSchema create a Sql type Timestamp with precision 0. So It is possible to lose precision when get timestamp data from Cassandra.

As for above two issues, Whether we should solve them at this PR or creat new jira issues for them?

Reply:
For problem 2. Even if C* adapter supports precision, in avatica, `DateTimeUtils` handles milliseconds timestamp as below and neglects the precision, so that the string returned losses precision.
```
 /** Helper for CAST({timestamp} AS VARCHAR(n)). */
  public static String unixTimestampToString(long timestamp) {
    return unixTimestampToString(timestamp, 0);
  }
```

Reply:
For timestamp types, the issue may exist in many places, including Avatica. So I'm inclined to open an umbrella JIRA and fix them.

Reply:
@DonnyZone Creating an umbrella issue to fix the adapter literal data types is a good idea, go ahead if you think that is necessary. @wenhuitang for this patch, let's just fix the type that has no precision, which does not bring in confusion.